### PR TITLE
Added setLeverage method to gateio

### PR DESCRIPTION
**Pull request by samgermain**
I'm getting this error currently, with or without setting `cross_leverage_limit`

```
gateio.setLeverage (0, BTC/USDT:USDT, [object Object])
fetch:
 gateio POST https://api.gateio.ws/api/v4/futures/usdt/positions/BTC_USDT/leverage 
Request:
 {
  KEY: '...',
  Timestamp: '1635164636',
  SIGN: '...',
  'Content-Type': 'application/json'
} 
 {"leverage":"0","cross_leverage_limit":"10"} 

handleRestResponse:
 gateio POST https://api.gateio.ws/api/v4/futures/usdt/positions/BTC_USDT/leverage 400 Bad Request 
Response:
 {
  Connection: 'keep-alive',
  'Content-Type': 'application/json',
  Date: 'Mon, 25 Oct 2021 12:23:56 GMT',
  Server: 'openresty',
  'Transfer-Encoding': 'chunked'
} 
 {"label":"INVALID_PROTOCOL","detail":"invalid argument: #3"}
 

BadRequest gateio invalid argument: #3
---------------------------------------------------
[BadRequest] gateio invalid argument: #3

    at handleErrors               js/gateio.js:2268                   throw new Error (this.id + ' ' + message);                                      
    at                            js/base/Exchange.js:657             this.handleErrors (response.status, response.statusText, url, method, responseH…
    at processTicksAndRejections  internal/process/task_queues.js:95                                                                                  
    at async timeout              js/base/functions/time.js:195       return await Promise.race ([promise, expires.then (() => { throw new TimedOut (…
    at setLeverage                js/gateio.js:2296                   return await this[method] (this.extend (request, params));                      
    at async main                 examples/js/cli.js:243              const result = await exchange[methodName] (... args)                            

gateio invalid argument: #3
Waiting for the debugger to disconnect...
```

Any insight?

### CODE REVIEW

File: js/gateio.js [hunk 1]
```diff
@@ -1217,6 +1169,43 @@ module.exports = class gateio extends Exchange {
...
+        }
+        const method = this.getSupportedMapping (market['type'], {
+            'swap': 'privateFuturesGetSettleAccountBook',
+            'futures': 'privateDeliveryGetSettleAccountBook',
+        });
+        const response = await this[method] (this.extend (request, params));
+        const result = [];
+        for (let i = 0; i < response.length; i++) {
+            const entry = response[i];
+            const timestamp = Precise.stringMul (entry['time'], '1000');
```
Comment (frosty00) [inline js/gateio.js:1195]: you can use `this.safeTimestamp` here

File: js/gateio.js [hunk 2]
```diff
@@ -1217,6 +1169,43 @@ module.exports = class gateio extends Exchange {
...
+            result.push ({
+                'info': entry,
+                'symbol': symbol,
+                'code': this.safeCurrencyCode (entry['text']),
```
Comment (frosty00) [inline js/gateio.js:1199]:
```
this.safeCurrencyCode (this.safeString (entry, 'text'))
````

File: js/gateio.js [hunk 3]
```diff
@@ -1217,6 +1169,43 @@ module.exports = class gateio extends Exchange {
...
+                'timestamp': timestamp,
+                'datetime': this.iso8601 (timestamp),
+                'id': undefined,
+                'amount': entry['change'],
```
Comment (frosty00) [inline js/gateio.js:1203]:
```
this.safeNumber (entry, 'change')
```

File: js/gateio.js [hunk 4]
```diff
@@ -2324,19 +2324,19 @@ module.exports = class gateio extends Exchange {
         const entirePath = '/' + type + endPart;
         let url = this.urls['api'][authentication] + entirePath;
         if (authentication === 'public') {
-            if (Object.keys (query).length > 0) {
+            if (Object.keys (query).length) {
```
Comment (samgermain) [inline js/gateio.js:2327]: Why would this be a fix?

Comment (kroitor) [reply js/gateio.js:2327]: It's due to how the transpiler treats that specific conditional:

- https://github.com/ccxt/ccxt/blob/master/build/transpile.js#L232
- https://github.com/ccxt/ccxt/blob/master/build/transpile.js#L394

Comment (kroitor) [reply js/gateio.js:2327]: We have to keep this type of conditional uniform across the lib. In Python and PHP, the same check looks a bit different syntactically, so it's just a special case for checking if the object is empty (has no keys).

**Pull request merged**